src/go/configgenerator/filterconfig/filter_generator.go

```diff
@@ -0,0 +1,262 @@
+			},
+		})
+	}
+
+	// Add JWT Authn filter if needed.
+	if !serviceInfo.Options.SkipJwtAuthnFilter {
+		// TODO(b/176432170): Handle errors here, prevent startup.
+		filterGenerators = append(filterGenerators, &FilterGenerator{
+			FilterName:            util.JwtAuthn,
+			FilterGenFunc:         jaFilterGenFunc,
+			PerRouteConfigGenFunc: jaPerRouteFilterConfigGen,
+		})
+	}
+
+	// Add Service Control filter if needed.
+	if !serviceInfo.Options.SkipServiceControlFilter {
+		filterGenerators = append(filterGenerators, &FilterGenerator{
+			FilterName:            util.ServiceControl,
+			FilterGenFunc:         scFilterGenFunc,
+			PerRouteConfigGenFunc: scPerRouteFilterConfigGen,
+		})
+	}
+
+	// Add gRPC Transcoder filter and gRPCWeb filter configs for gRPC backend.
+	if serviceInfo.GrpcSupportRequired {
+		// grpc-web filter should be before grpc transcoder filter.
+		// It converts content-type application/grpc-web to application/grpc and
+		// grpc transcoder will bypass requests with application/grpc content type.
+		// Otherwise grpc transcoder will try to transcode a grpc-web request which
+		// will fail.
+		filterGenerators = append(filterGenerators, &FilterGenerator{
+			FilterName: util.GRPCWeb,
+			FilterGenFunc: func(sc *ci.ServiceInfo) (*hcmpb.HttpFilter, []*ci.MethodInfo, error) {
+				return &hcmpb.HttpFilter{
+					Name: util.GRPCWeb,
+				}, nil, nil
+			},
+		})
+
+		filterGenerators = append(filterGenerators, &FilterGenerator{
+			FilterName: util.GRPCJSONTranscoder,
+			FilterGenFunc: func(sc *ci.ServiceInfo) (*hcmpb.HttpFilter, []*ci.MethodInfo, error) {
+				return makeTranscoderFilter(serviceInfo), nil, nil
+			},
+		})
+	}
+
+	filterGenerators = append(filterGenerators, &FilterGenerator{
+		FilterName:            util.BackendAuth,
+		FilterGenFunc:         baFilterGenFunc,
+		PerRouteConfigGenFunc: baPerRouteFilterConfigGen,
+	})
+
+	filterGenerators = append(filterGenerators, &FilterGenerator{
+		FilterName:            util.PathRewrite,
+		FilterGenFunc:         prFilterGenFunc,
+		PerRouteConfigGenFunc: prPerRouteFilterConfigGen,
+	})
+
+	if serviceInfo.Options.EnableGrpcForHttp1 {
+		// Add GrpcMetadataScrubber filter to retain gRPC trailers
+
+		filterGenerators = append(filterGenerators, &FilterGenerator{
+			FilterName: util.GrpcMetadataScrubber,
+			FilterGenFunc: func(sc *ci.ServiceInfo) (*hcmpb.HttpFilter, []*ci.MethodInfo, error) {
+				return &hcmpb.HttpFilter{
+					Name: util.GrpcMetadataScrubber,
+				}, nil, nil
+			},
+		})
+	}
+
+	// Add Envoy Router filter so requests are routed upstream.
+	// Router filter should be the last.
+	filterGenerators = append(filterGenerators, &FilterGenerator{
+		FilterName: util.Router,
+		FilterGenFunc: func(sc *ci.ServiceInfo) (*hcmpb.HttpFilter, []*ci.MethodInfo, error) {
+			return makeRouterFilter(serviceInfo.Options), nil, nil
+		},
+	})
+	return filterGenerators, nil
+}
+
+func makeTranscoderFilter(serviceInfo *ci.ServiceInfo) *hcmpb.HttpFilter {
```

Comment:
Move these helper functions into different files.

If we have a different version of MakeFilterGenerator function in g3, we don't want to rely on copybara regex to replace the entire function and add imports.

Instead we should just have a forked copy of this file in g3. It will be easy to do this if these helper functions are in different files, then we only need to worry about diffs in `MakeFilterGenerator`.

Reply:
ACK